dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/metrics/BatchMetricsDispatcher.kt

```diff
@@ -47,7 +48,7 @@ internal class BatchMetricsDispatcher(
     }
 
     override fun sendBatchClosedMetric(batchFile: File, batchMetadata: BatchClosedMetadata) {
-        if (trackName == null || !sampler.sample()) {
+        if (!batchFile.existsSafe(internalLogger) || trackName == null || !sampler.sample()) {
```

Comment:
small performance gain - let's put the check if file exists to the right - this way we won't do it if metric is not sampled, we are saving some I/O capacity and block the thread less

```suggestion
        if (trackName == null || !sampler.sample() || !batchFile.existsSafe(internalLogger)) {
```

Reply:
Good catch